src/courtroom.cpp

```diff
@@ -1339,8 +1354,14 @@ void Courtroom::handle_chatmessage(QStringList *p_contents)
   append_ic_text(": " + m_chatmessage[MESSAGE], f_showname);
 
   previous_ic_message = f_message;
-
-  int objection_mod = m_chatmessage[OBJECTION_MOD].toInt();
+  bool ok;
+  int objection_mod = m_chatmessage[OBJECTION_MOD].toInt(&ok, 10); //checks if its a custom obj.
+  objection_custom = ""; //added this for safety reasons
+  if(ok == false)
+  {
+      objection_mod = 4;
+      objection_custom = m_chatmessage[OBJECTION_MOD].split('&')[1]; //takes the name of custom objection
```

Comment:
May suffer out-of-bounds error, causing a crash.

Reply:
For which array are you talking about

Reply:
The expression `m_chatmessage[OBJECTION_MOD].split('&')[1]`

Reply:
The only scenarios which will cause out of bound errors are from either somebody modifying the server so there is no packet checking (one of the reasons `If '4'  in button and not '<and>' in button` exists) and somebody modified the client so it doesn't send '&' when appropriate

Reply:
Incoming network packets generally should not be trusted. The server sanitizing network input is an expectation, but not a guarantee.